stdlib/format.ml

```diff
@@ -815,29 +815,48 @@ let pp_set_margin state n =
 (** Geometry functions and types *)
 type geometry = { max_indent:int; margin: int}
 
+let validate_geometry {margin; max_indent} =
+  if max_indent < 2 then
+    Error "max_indent < 2"
+  else if margin <= max_indent then
+    Error "margin <= max_indent"
+  else Ok ()
+
 let check_geometry geometry =
-  geometry.max_indent > 1
-  &&  geometry.margin > geometry.max_indent
+  match validate_geometry geometry with
+  | Ok () -> true
+  | Error _ -> false
 
 let pp_get_margin state () = state.pp_margin
 
+let pp_set_full_geometry state {margin; max_indent} =
```

Comment:
I am not sure if `full` is the right adjective here: maybe `unsafe`? Or an adverb `directly`?

Reply:
I'll leave it as-is: this is not exposed in the public API, and the `_full` prefix will become self-evident after we start adding extra record fields, while `pp_set_geometry` is left partial due to backward-compatibility constraints.